### PR TITLE
[NTUSER] Fix a 64 bit bug in timer code

### PR DESCRIPTION
**Pull request by tkreuzer**
The return value of RtlFindClearBitsAndSet is an ULONG, assigning it to an ULONG_PTR will not sign extend it. The error value will stay 0xFFFFFFFF. Comparing it to (UINT_PTR)-1 will sign extend and thus compare it to 0xFFFFFFFFFFFFFFFF on x64. Also use NUM_WINDOW_LESS_TIMERS to initialize the bitmap, rather than the calculated size. This does not make a difference with the current value (32768), but if it was not the case, the bitmap would be larger than this, resulting in invalid bitmap indices being returned, which would cause bugs later on.

### CODE REVIEW

**Comment by binarymaster** [comment]
Does this interfere with https://github.com/reactos/reactos/pull/7277 ?

Also please change `[WIN32K]` prefix to `[NTUSER]` for consistency.

File: win32ss/user/ntuser/timer.c [hunk 1]
```diff
@@ -221,22 +222,22 @@ IntSetTimer( PWND Window,
   {
       IntLockWindowlessTimerBitmap();
 
-      IDEvent = RtlFindClearBitsAndSet(&WindowLessTimersBitMap, 1, HintIndex++);
-      if (IDEvent == (UINT_PTR)-1)
+      ulBitmapIndex = RtlFindClearBitsAndSet(&WindowLessTimersBitMap, 1, HintIndex++);
+      if (ulBitmapIndex == 0xFFFFFFFF)
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/timer.c:226]:
```suggestion
      if (ulBitmapIndex == ULONG_MAX)
```

File: win32ss/user/ntuser/timer.c [hunk 2]
```diff
@@ -221,22 +222,22 @@ IntSetTimer( PWND Window,
...
       {
          HintIndex = HINTINDEX_BEGIN_VALUE;
-         IDEvent = RtlFindClearBitsAndSet(&WindowLessTimersBitMap, 1, HintIndex++);
+         ulBitmapIndex = RtlFindClearBitsAndSet(&WindowLessTimersBitMap, 1, HintIndex++);
       }
-      if (IDEvent == (UINT_PTR) -1)
+      if (ulBitmapIndex == 0xFFFFFFFF)
```
Comment (SergeGautherie) [inline win32ss/user/ntuser/timer.c:231]:
```suggestion
      if (ulBitmapIndex == ULONG_MAX)
```

**Review by Doug-Lyons — APPROVED** [review]

File: win32ss/user/ntuser/timer.c [hunk 3]
```diff
@@ -184,6 +184,7 @@ IntSetTimer( PWND Window,
 {
   PTIMER pTmr;
   UINT Ret = IDEvent;
+  ULONG ulBitmapIndex;
```
Comment (HBelusca) [inline win32ss/user/ntuser/timer.c:187]: Not sure the "ul" prefix is needed.

File: win32ss/user/ntuser/timer.c [hunk 4]
```diff
@@ -184,6 +184,7 @@ IntSetTimer( PWND Window,
 {
   PTIMER pTmr;
   UINT Ret = IDEvent;
```
Comment (HBelusca) [inline win32ss/user/ntuser/timer.c:186]: Why does the Ret variable is UINT, but it is used as the returned value of IntSetTimer which returns an `UINT_PTR`?
(and it is assigned initially to IDEvent which is an `UINT_PTR`...) Shouldn't this need to be fixed?

Note that this function corresponds to the win32 [`SetTimer()`](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-settimer) which indeed uses `UINT_PTR` for these.

Comment (tkreuzer) [reply win32ss/user/ntuser/timer.c]: Changed it to UINT_PTR.

**Review by JoachimHenze — APPROVED** [review]
Approved. If they don't hurt your feelings, I do think the 2 Serge-proposals are good, since they might protect from accidentally typing one F too much or too little. Ftr: I don't think that using the macro instead would reduce the readability.

**Pull request merged**